algorithms/integration/kapton_2019_correction.py

```diff
@@ -242,6 +257,85 @@ def _check_int_edge_pts(int_edge_pts):
                 return new_int_edge_pts
 
+        def _orient_det_edge_pts(det_edge_pts):
+            """
+            Function to orient the detector edge points counterclockwise
+            such that the first edge point index is in the upper left quadrant
+            (i.e., -x, +y). This isn't explictly necessary but nevertheless useful
+            b/c the function abs_bounding_lines_on_image always assumes that the
+            detector edges used to help calculate the Kapton absorption max and edge
+            are a parallel to the y-axis. This function enforces that a convention
+            so that this assumption in the later logic will be true.
+            """
+            assert len(det_edge_pts) == 4, "Detectors can only be defined by 4 points"
+            n_edges = float(len(det_edge_pts))
+            # find center of detector
+            x = [p[0] for p in det_edge_pts]
+            y = [p[1] for p in det_edge_pts]
+            z = [p[2] for p in det_edge_pts]
+            center = [sum(x) / n_edges, sum(y) / n_edges, sum(z) / n_edges]
+            # define reference vectors used to determine rel. orientation
+            x_ref = [center[0] + 100, center[1], center[2]]
+            y_ref = [center[0], center[1] + 100, center[2]]
+            x_ref_vec = (col(x_ref) - col(center)).normalize()
+            y_ref_vec = (col(y_ref) - col(center)).normalize()
+            # determine orientation of each edge pt wrt x and y axis
+            x_axis_angles = []
+            y_axis_angles = []
+            for edge in det_edge_pts:
+                vec = (col(edge) - col(center)).normalize()
+                x_axis_angles.append(np.degrees(np.arccos(vec.dot(x_ref_vec))))
+                y_axis_angles.append(np.degrees(np.arccos(vec.dot(y_ref_vec))))
+            x_axis_angles = [
+                (90.0 - xang) if xang > 90.0 else xang for xang in x_axis_angles
+            ]
+            y_axis_angles = [
+                (90.0 - yang) if yang > 90.0 else yang for yang in y_axis_angles
+            ]
+            # assign edge pt indices based on quadrant position starting in 4th quadrant (-x.-y)
+            # proceeding counterclockwise
+            edge_0 = np.where(
+                (np.asarray(x_axis_angles) < 0) & (np.asarray(y_axis_angles) > 0)
+            )[0][0]
+            edge_1 = np.where(
+                (np.asarray(x_axis_angles) < 0) & (np.asarray(y_axis_angles) < 0)
+            )[0][0]
+            edge_2 = np.where(
+                (np.asarray(x_axis_angles) > 0) & (np.asarray(y_axis_angles) < 0)
+            )[0][0]
+            edge_3 = np.where(
+                (np.asarray(x_axis_angles) > 0) & (np.asarray(y_axis_angles) > 0)
+            )[0][0]
+
+            return [
+                det_edge_pts[edge_0],
+                det_edge_pts[edge_1],
+                det_edge_pts[edge_2],
+                det_edge_pts[edge_3],
+            ]
+
+        def _make_kapton_distance_vec(
```

Comment:
Utility function used to help define the relative location of a given kapton intersection edge point relative to the detector center. This is useful when determining how to order the kapton intersection edge points when returning out of the abs_bounding_lines_in_mm function